### PR TITLE
Update-Help test updates

### PR DESCRIPTION
**Pull request by anmenaga**
## PR Summary

Currently building PS is required on a test machine for a test pass to succeed.
One of the test work items is to decouple test passes from build requirement; i.e. tests should be runnable on released PS packages.
This PR fixes test code (Update-Help tests) that currently fail on released PS packages.

Test results (`test\powershell\engine\Help`) before fix:
beforefix

Test results (`test\powershell\engine\Help`) after fix:
afterfix

## PR Checklist

Note: Please mark anything not applicable to this PR `NA`.

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - [x] Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [NA] User facing [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - [ ] Issue filed - Issue link:
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [NA] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
    - [ ] [Add `[feature]` if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready.

### CODE REVIEW

File: test/powershell/engine/Help/HelpSystem.Tests.ps1 [hunk 1]
```diff
@@ -64,6 +76,28 @@ function RunTestCase
...
+    try
+    {
+        $filepath = Join-Path $Path "testfile-deleteme.txt"
+        "test" | Out-File $filepath
+    }
+    catch
+    {
+        $readonly = $_.Exception.GetType().Name -eq 'UnauthorizedAccessException'
+    }
+    Remove-Item -Path $filepath -ErrorAction SilentlyContinue
```
Comment (iSazonov) [inline test/powershell/engine/Help/HelpSystem.Tests.ps1]: Please add `-Force`.

File: test/powershell/engine/Help/HelpSystem.Tests.ps1 [hunk 2]
```diff
@@ -5,6 +5,18 @@ function UpdateHelpFromLocalContentPath
 {
     param ([string]$ModuleName, [string]$Tag = 'CI')
 
+    # Update-Help fails if module path is Not writable, so skip tests in this situation
```
[the remark this answers is not in the document]
Comment (adityapatwardhan) [reply test/powershell/engine/Help/HelpSystem.Tests.ps1]: Yes, we do not run the tests under sudo on Linux.

File: test/powershell/engine/Help/UpdatableHelpSystem.Tests.ps1 [hunk 3]
```diff
@@ -133,6 +133,27 @@ $testCases = @{
 $modulesInBox = @("Microsoft.PowerShell.Core"
                   Get-Module -ListAvailable | ForEach-Object{$_.Name}
 )
+function PathIsReadOnly
```
Comment (adityapatwardhan) [inline test/powershell/engine/Help/UpdatableHelpSystem.Tests.ps1]: Can we move this function to a common file and then dot source it in both the test files.

File: test/powershell/engine/Help/UpdatableHelpSystem.Tests.ps1 [hunk 4]
```diff
@@ -174,8 +195,20 @@ function RunUpdateHelpTests
...
+            {
+                $skip = $MicrosoftPowerShellCorePathIsReadOnly
+            }
+            else
+            {
+                $modulePath = (Get-Module -Name $moduleName -ListAvailable).ModuleBase
+                $skip = PathIsReadOnly $modulePath
+            }
+
+            # -Skip and -$Pending are mutually exclusive on It block, so skip Pending tests
```
Comment (adityapatwardhan) [inline test/powershell/engine/Help/UpdatableHelpSystem.Tests.ps1]: Skip and Pending have specific meanings. you can use splatting to choose one of them for the It block.

**Review by adityapatwardhan — CHANGES_REQUESTED** [review]

File: test/powershell/engine/Help/HelpSystem.Tests.ps1 (hunk 2 above)
[the remark this answers is not in the document]
Comment (iSazonov) [reply test/powershell/engine/Help/HelpSystem.Tests.ps1]: Perhaps we should not complicate these tests if we want to implement #5645?

[the remark this answers is not in the document]
Comment (adityapatwardhan) [reply test/powershell/engine/Help/HelpSystem.Tests.ps1]: I believe we need to go through a bunch to tests when #5645 is done. We can revert these changes then.

**Comment by adityapatwardhan** [comment]
Now that RequiresSudoOnUnix is available. We can simplify the logic over here. Please have a look at: https://github.com/PowerShell/PowerShell/pull/6145

**Comment by adityapatwardhan** [comment]
I believe https://github.com/PowerShell/PowerShell/pull/6223 should fix the issue this PR is trying to fix. 
@anmenaga if you agree, please close the PR.

**Comment by anmenaga** [comment]
this is covered by #6223 .

**Pull request closed without merging**